### ISSUE TITLE
Use LazyCache instead of MemoryCache

### ISSUE THREAD

**Issue by RomanovM** [nopSolutions/nopCommerce#5173]
nopCommerce version: develop

Let's look at LazyCache, it seems quite more performant than the current native MemoryCache implementation.

Source: https://blog.novanet.no/asp-net-core-memory-cache-is-get-or-create-thread-safe/

It makes sense to look after #2287

**Comment by AndreiMaz** [comment]
decided to postpone as we haven't seen much performance enhancement
